### ISSUE TITLE
RFE: add name parameters to list_* methods

### ISSUE THREAD

**Issue by ktdreyer** [apache/libcloud#1542]
## Feature Request

In large environments, running `list_nodes()` and then filtering the list by name client-side is expensive.

It would be fantastic if I could filter nodes, images, and volumes server-side instead. Can we add a new `name` kwarg to `list_nodes()`, `list_images()`, `list_volumes()`, etc?

**Comment by Kami** [comment]
I agree that this would be a good feature to have. Some drivers already implement this functionality via extension / ex_ arguments.

As far as implementation goes - ideally we should filter on the server side where that is supported, but for compatibility reasons, where it's not supported, we can implement simple late filtering on the client side.
